File: ruby/invoice_item_generator.rb
require './invoice_item.rb'

class InvoiceItemGenerator
  # TODO: Use up the balance in the account to create reimbursement InvoiceItems. Once
  # the account is empty, create overage InvoiceItems for any additional spending.
  # For example:
  #   Account balance: $100
  #   Transaction amounts: $50, $50, $5
  #   InvoiceItems: $50 reimbursement, $50 reimbursement, $5 overage
  def self.generate_for_transactions(account, transactions)
    []
  end
end

File: dotnet/MiniThatch/InvoiceItemGenerator.cs
namespace MiniThatch;
using System.Collections.Generic;

public static class InvoiceItemGenerator
{
    public static List<InvoiceItem> GenerateForTransactions(Account account, List<Transaction> transactions)
    {
        /*
         * TODO: Use the balance in the account to create reimbursement InvoiceItems. once
         * the account is empty, create overage InvoiceItems for the remaining transaction amounts.
         * For example:
         *   Account balance: $100
         *   Transaction amounts: $50, $80, $5
         *   InvoiceItems: $50 reimbursement, $50 reimbursement, $30 overage, $5 overage
         */
        return new List<InvoiceItem>();
    }
}